tensorflow_probability/python/distributions/moyal.py

```diff
@@ -0,0 +1,240 @@
+      dtype_util.assert_same_float_dtype([loc, scale])
+      # Positive scale is asserted by the incorporated Moyal bijector.
+      self._moyal_bijector = moyal_cdf_bijector.MoyalCDF(
+          loc=loc, scale=scale, validate_args=validate_args)
+
+      # Because the uniform sampler generates samples in `[0, 1)` this would
+      # cause samples to lie in `(inf, -inf]` instead of `(inf, -inf)`. To fix
+      # this, we use `np.finfo(dtype_util.as_numpy_dtype(self.dtype).tiny`
+      # because it is the smallest, positive, 'normal' number.
+      batch_shape = distribution_util.get_broadcast_shape(loc, scale)
+      super(Moyal, self).__init__(
+          # TODO(b/137665504): Use batch-adding meta-distribution to set the
+          # batch shape instead of tf.ones.
+          distribution=uniform.Uniform(
+              low=np.finfo(dtype_util.as_numpy_dtype(dtype)).tiny,
+              high=tf.ones(batch_shape, dtype=dtype),
+              allow_nan_stats=allow_nan_stats),
+          # The Moyal bijector encodes the CDF function as the forward,
+          # and hence needs to be inverted.
+          bijector=invert_bijector.Invert(
+              self._moyal_bijector, validate_args=validate_args),
+          parameters=parameters,
+          name=name)
+
+  @staticmethod
+  def _param_shapes(sample_shape):
+    return dict(
+        zip(('loc', 'scale'),
+            ([tf.convert_to_tensor(sample_shape, dtype=tf.int32)] * 2)))
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(loc=0, scale=0)
+
+  @property
+  def loc(self):
+    """Distribution parameter for the location."""
+    return self._moyal_bijector.loc
+
+  @property
+  def scale(self):
+    """Distribution parameter for scale."""
+    return self._moyal_bijector.scale
+
+  def _entropy(self):
+    # Use broadcasting rules to calculate the full broadcast sigma.
+    scale = self.scale * tf.ones_like(self.loc)
+    return (1/2 * (1. + 2 * tf.math.log(scale) +
+                   np.euler_gamma + np.log(4. * np.pi)))
+
+  def _log_prob(self, x):
+    scale = tf.convert_to_tensor(self.scale)
+    z = (x - self.loc) / scale
+    return - 1/2 * (z + tf.exp(-z)) - tf.math.log(np.sqrt(2. * np.pi) * scale)
+
+  def _mean(self):
+    return self.loc + self.scale * (np.euler_gamma + np.log(2.))
+
+  def _stddev(self):
+    return self.scale * tf.ones_like(self.loc) * np.pi / np.sqrt(2.)
+
+  def _mode(self):
+    return self.loc * tf.ones_like(self.scale)
+
+  def _default_event_space_bijector(self):
+    # TODO(b/145620027) Finalize choice of bijector. Consider switching to
+    # Chain([Softplus(), Log()]) to lighten the doubly-exponential right tail.
+    return identity_bijector.Identity(validate_args=self.validate_args)
+
+  def _parameter_control_dependencies(self, is_init):
+    return self._moyal_bijector._parameter_control_dependencies(is_init)  # pylint: disable=protected-access
+
+
+@kullback_leibler.RegisterKL(Moyal, Moyal)
+def _kl_moyal_moyal(a, b, name=None):
```

Comment:
Can you add a reference for this?

Reply:
I solved this with Mathematica, haven't found a reference as this distribution is not exactly widely used...

Reply:
Can you add an inline comment specifying that (that this was computed via Mathematica)?